modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/inbound/InboundAndesChannelEvent.java

```diff
@@ -63,9 +60,17 @@ public InboundAndesChannelEvent( UUID channelID) {
     @Override
     public void updateState() throws AndesException {
         switch (eventType) {
-            case CHANNEL_OPEN_EVENT:
-                break;
-            case CHANNEL_CLOSE_EVENT:
+            case CHANNEL_RECOVER_EVENT:
+                AndesSubscription subscription  = AndesContext.getInstance().
+                        getAndesSubscriptionManager().getSubscriptionByProtocolChannel(channelID);
+
+                // Subscription can be null if we send a recover call without subscribing. In this case we do not have
+                // to recover any messages. Therefore print a log and return.
+                if (null == subscription) {
+                    log.warn("Cannot handle recover. No subscriptions found for channel " + channelID);
```

Comment:
Good question. But subscription might have closed in a sudden. In such cases this  can happen (by design).